R/furness_balancing.R

```diff
@@ -0,0 +1,304 @@
+# Furness balancing functions
+
+#' Furness Balance a matrix
+#'
+#' Furness balancing fills a matrix when only the row and column sums are known. 
+#'
+#' @param mat matrix of numeric values e.g. `matrix(rep(1, 6), nrow = 3)`
+#' @param rsum vector of desired row sums without NAs
+#' @param csum vector of desired column sums without NAs
+#' @param n number of iterations of balancing
+#' @param check logical (default TRUE) check if output is valid
+#' @param int_only logical (default FALSE) only return whole numbers
+#' @param quiet logical if FALSE extra messages
+#' @return A vector of rounded numeric
+#' @export
+#' @example 
+#' mat = matrix(rep(1, 6), nrow = 3) 
```

Comment:
Great to see an example in the package's style. Clearly works but I think we should have an example that uses real example data. Could example datasets in this package work, or do we need to create more packaged data, e.g. starting with NTEM? Having done some reading it seems to me that Furness balancing is more about updating SIM estimates over time, which would be v. handy and links nicely with the spanishoddata package.

Reply:
Don't know about over time, all of these functions are for filling in matrices where you know row/column totals but not internal values. So classic example is knowing the number of pupils at schools and the number of children in each LSOA, but you don't have OD data. You can prime the matrix with values from a distance decay curve and then run the furness balance and you should get a reasonable OD matrix.

Reply:
Doesn't the function assume OD data in matrix form, with rows representing origins and columns representing destinations?

Fine if so, although will not work on large sparse OD datasets, e.g. all schools to all LSOAs which would be 10 * 30 = 300,000,000 cells.

Reply:
Yes all these functions a matrix based

Reply:
OK, need to chat this over. Next time both in office could be good.